initial-ai-src/script.js

```diff
@@ -0,0 +1,587 @@
+      case 30:
+        return 30;
+      case 35:
+        return 35;
+      case 40:
+        return 40;
+      default:
+        return 0;
+    }
+  }
+
+  calculateDiscountBreakdown(
+    totalTickets,
+    ticketPrice,
+    currentTicketCount = 0
+  ) {
+    const breakdown = [];
+    let ticketsProcessed = 0;
+
+    while (ticketsProcessed < totalTickets) {
+      const absoluteTicketNumber = currentTicketCount + ticketsProcessed + 1;
+      const discount = this.getDiscountPercentage(absoluteTicketNumber);
+
+      // Find how many tickets at this discount level
+      let ticketsAtThisDiscount = 1;
+      let nextTicketNumber = absoluteTicketNumber + 1;
+
+      while (
+        ticketsProcessed + ticketsAtThisDiscount < totalTickets &&
+        this.getDiscountPercentage(nextTicketNumber) === discount
+      ) {
+        ticketsAtThisDiscount++;
+        nextTicketNumber++;
+      }
+
+      const tierStart = ticketsProcessed + 1;
+      const tierEnd = ticketsProcessed + ticketsAtThisDiscount;
+
+      breakdown.push({
+        range: `${tierStart}-${tierEnd}`,
+        tickets: ticketsAtThisDiscount,
+        discount: discount,
+        pricePerTicket: ticketPrice * (1 - discount / 100),
+        totalCost: ticketsAtThisDiscount * ticketPrice * (1 - discount / 100),
+        absoluteRange: `${absoluteTicketNumber}-${
+          absoluteTicketNumber + ticketsAtThisDiscount - 1
+        }`,
+      });
+
+      ticketsProcessed += ticketsAtThisDiscount;
+    }
+
+    return breakdown;
+  }
+
+  calculate() {
+    const tripsPerWeek =
+      parseInt(document.getElementById("trips-per-week").value) || 0;
+    const period = parseInt(
+      document.getElementById("calculation-period").value
+    );
+    const selectedZone = document.getElementById("zone-selection").value;
+    const currentDiscount = parseInt(
+      document.getElementById("current-discount").value
+    );
+
+    // Get zone-specific ticket price and current ticket count
+    const ticketPrice = this.zonePrices[selectedZone];
+    const currentTicketCount = this.getCurrentTicketCount(currentDiscount);
+
+    // Calculate total trips for the period
+    const weeksInPeriod = period / 7;
+    const totalTrips = Math.round(tripsPerWeek * weeksInPeriod);
+
+    if (totalTrips === 0) {
+      this.displayResults({
+        totalTrips: 0,
+        totalCost: 0,
+        totalSavings: 0,
+        averageDiscount: 0,
+        breakdown: [],
+        selectedZone,
+        currentDiscount,
+        currentTicketCount,
+      });
+      this.displayComparison([], 0, period);
+      this.displayBreakEvenAnalysis(selectedZone, currentTicketCount, period);
+      return;
+    }
+
+    // Calculate breakdown starting from current discount level
+    const breakdown = this.calculateDiscountBreakdown(
+      totalTrips,
+      ticketPrice,
+      currentTicketCount
+    );
+
+    // Calculate totals
+    const totalCost = breakdown.reduce((sum, tier) => sum + tier.totalCost, 0);
```

Comment:
🎨 I generally avoid `reduce` because it confuses people, but this use seems reasonable to me